<a href="https://colab.research.google.com/github/fatmakara/PythonVeriAnaliz/blob/main/Python3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <center> Güney Marmara ve TRAI
# <center> Python Veri Analizi Eğitimi
# <center> Ödev 3

### 2021 EUR/TRY kur tablosunun incelenmesi

1. <i> Yahoo Finance</i> sitesinden 2021 yılına ait EUR/TRY kur değerlerini indiriniz

In [1]:
import yfinance as yf
import plotly.express as px
import plotly.graph_objects as go
import pandas as pd
import matplotlib.pyplot as plt
from plotly.subplots import make_subplots
import warnings
warnings.filterwarnings('ignore')


In [2]:
kur_2021= yf.download('EURTRY=X', start ='2021-01-01', end='2021-12-31')
kur_2021

[*********************100%***********************]  1 of 1 completed


Open      High       Low     Close  Adj Close  Volume
Date                                                                 
2021-01-01   9.08470   9.08470   9.08470   9.08470    9.08470       0
2021-01-04   9.11047   9.11154   8.99617   9.10568    9.10568       0
2021-01-05   9.09273   9.11239   9.04298   9.08882    9.08882       0
2021-01-06   9.08044   9.09500   8.92864   9.08177    9.08177       0
2021-01-07   9.00475   9.07176   8.88154   9.00614    9.00614       0
...              ...       ...       ...       ...        ...     ...
2021-12-24  12.79533  13.67650  12.51641  12.79752   12.79752       0
2021-12-27  12.28318  13.03196  12.16917  12.23056   12.23056       0
2021-12-28  12.96156  13.47629  12.88283  13.05254   13.05254       0
2021-12-29  13.35232  14.37038  13.31566  13.35252   13.35252       0
2021-12-30  14.32090  15.11444  14.05750  14.31055   14.31055       0

[260 rows x 6 columns]

2. Veriyi özet metodları ile inceleyiniz

In [4]:
kur_2021.head(10)

Open     High      Low    Close  Adj Close  Volume
Date                                                             
2021-01-01  9.08470  9.08470  9.08470  9.08470    9.08470       0
2021-01-04  9.11047  9.11154  8.99617  9.10568    9.10568       0
2021-01-05  9.09273  9.11239  9.04298  9.08882    9.08882       0
2021-01-06  9.08044  9.09500  8.92864  9.08177    9.08177       0
2021-01-07  9.00475  9.07176  8.88154  9.00614    9.00614       0
2021-01-08  8.98835  9.05862  8.95726  8.99850    8.99850       0
2021-01-11  8.96589  9.13313  8.95741  8.96309    8.96309       0
2021-01-12  9.08772  9.12800  9.02964  9.09150    9.09150       0
2021-01-13  9.09178  9.12552  8.99663  9.09140    9.09140       0
2021-01-14  8.99276  9.04875  8.91476  9.00060    9.00060       0

In [5]:
kur_2021.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 260 entries, 2021-01-01 to 2021-12-30
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Open       260 non-null    float64
 1   High       260 non-null    float64
 2   Low        260 non-null    float64
 3   Close      260 non-null    float64
 4   Adj Close  260 non-null    float64
 5   Volume     260 non-null    int64  
dtypes: float64(5), int64(1)
memory usage: 14.2 KB


In [6]:
kur_2021.describe()

Open        High         Low       Close   Adj Close  Volume
count  260.000000  260.000000  260.000000  260.000000  260.000000   260.0
mean    10.434622   10.573510   10.347226   10.439140   10.439140     0.0
std      1.754535    1.940096    1.688033    1.753083    1.753083     0.0
min      8.386340    8.464750    8.354830    8.387260    8.387260     0.0
25%      9.621848    9.695410    9.550110    9.629247    9.629247     0.0
50%     10.088800   10.146105   10.047840   10.089370   10.089370     0.0
75%     10.439128   10.471000   10.358152   10.442483   10.442483     0.0
max     18.671040   20.733000   18.657669   18.655109   18.655109     0.0

3. Veri temizleme: <br>
	• Tabloda sadece "Adj Close" (düzeltilmiş kapanış) sütununu bırakınız.<br>
	• "Adj Close" sütununu " Kapanış (TRY)" olarak yeniden adlandırınız.

In [7]:
kur_2021 = kur_2021[['Adj Close']]
kur_2021

Adj Close
Date                 
2021-01-01    9.08470
2021-01-04    9.10568
2021-01-05    9.08882
2021-01-06    9.08177
2021-01-07    9.00614
...               ...
2021-12-24   12.79752
2021-12-27   12.23056
2021-12-28   13.05254
2021-12-29   13.35252
2021-12-30   14.31055

[260 rows x 1 columns]

In [8]:
kur_2021.columns=['kapanış (TRY)']
kur_2021

kapanış (TRY)
Date                     
2021-01-01        9.08470
2021-01-04        9.10568
2021-01-05        9.08882
2021-01-06        9.08177
2021-01-07        9.00614
...                   ...
2021-12-24       12.79752
2021-12-27       12.23056
2021-12-28       13.05254
2021-12-29       13.35252
2021-12-30       14.31055

[260 rows x 1 columns]

4. Tabloya yeni sütun ekleme:<br>
	• Tabloya yeni bir sütun ekleyerek Kapanış (TRY) değerlerini 1 Ocak 2021 tarihindeki kapanış değerine göre normalize ediniz.<br>
	• BIST30 endeksinin 2021 yılı boyunca değişimini yorumlayınız.

In [11]:
kur_2021 = kur_2021[kur_2021['kapanış (TRY)'] > 0] #warning almamak için
kur_2021['Normalize kapanış (TRY)'] = kur_2021['kapanış (TRY)'] / kur_2021.iloc[0, 0]
kur_2021

kapanış (TRY)  Normalize kapanış (TRY)
Date                                              
2021-01-01        9.08470                 1.000000
2021-01-04        9.10568                 1.002309
2021-01-05        9.08882                 1.000454
2021-01-06        9.08177                 0.999678
2021-01-07        9.00614                 0.991353
...                   ...                      ...
2021-12-24       12.79752                 1.408689
2021-12-27       12.23056                 1.346281
2021-12-28       13.05254                 1.436761
2021-12-29       13.35252                 1.469781
2021-12-30       14.31055                 1.575236

[260 rows x 2 columns]

In [15]:
b30_21 = yf.download('XU030.IS', start = '2021-01-01', end = '2022-01-01')
b30_21 = b30_21[['Adj Close']]
b30_21.columns = ['BIST30']
b30_21['Normalize BIST30'] = b30_21['BIST30'] / b30_21.iloc[0, 0]
b30_21


[*********************100%***********************]  1 of 1 completed


BIST30  Normalize BIST30
Date                                     
2021-01-04  1652.900024          1.000000
2021-01-05  1640.400024          0.992438
2021-01-06  1657.699951          1.002904
2021-01-07  1678.199951          1.015306
2021-01-08  1694.400024          1.025107
...                 ...               ...
2021-12-27  2089.399902          1.264081
2021-12-28  2032.900024          1.229899
2021-12-29  2086.300049          1.262206
2021-12-30  2040.199951          1.234315
2021-12-31  2025.000000          1.225119

[239 rows x 2 columns]

01-01-2021 tarihinde 1.000000 değerinde başlamış ve 30-12-2021 tarihinde 1.575236 değerinde ulaşarak %57 lik bir artış meydana gelmiştir.  Haziran ve temmuz ayında en düşük değerlere ulaşmıştır.


5. Veriyi grafik olarak inceleyiniz:<br>
	• Tabloyu px grafiği olarak görselleştiriniz.<br>
	• Tabloyu go. grafiği olarak görselleştiriniz.

In [18]:
b30kur_2021= b30_21.merge(kur_2021, left_index = True, right_index = True)
#burada kapanış normalize 1'den başlamıyor.... 1.1.2021 değeri yok çünkü...
b30kur_2021['Normalize kapanış (TRY)'] = b30kur_2021['kapanış (TRY)'] / b30kur_2021.iloc[0, 2]
b30kur_2021

BIST30  Normalize BIST30  kapanış (TRY)  \
Date                                                       
2021-01-04  1652.900024          1.000000        9.10568   
2021-01-05  1640.400024          0.992438        9.08882   
2021-01-06  1657.699951          1.002904        9.08177   
2021-01-07  1678.199951          1.015306        9.00614   
2021-01-08  1694.400024          1.025107        8.99850   
...                 ...               ...            ...   
2021-12-24  2086.800049          1.262508       12.79752   
2021-12-27  2089.399902          1.264081       12.23056   
2021-12-28  2032.900024          1.229899       13.05254   
2021-12-29  2086.300049          1.262206       13.35252   
2021-12-30  2040.199951          1.234315       14.31055   

            Normalize kapanış (TRY)  
Date                                 
2021-01-04                 1.000000  
2021-01-05                 0.998148  
2021-01-06                 0.997374  
2021-01-07                 0.989068  
2021-01-08                 0.988229  
...                             ...  
2021-12-24                 1.405444  
2021-12-27                 1.343179  
2021-12-28                 1.433450  
2021-12-29                 1.466395  
2021-12-30                 1.571607  

[238 rows x 4 columns]

In [19]:
#px grafiği
px.line(b30kur_2021, y = ['Normalize BIST30', 'Normalize kapanış (TRY)'],
        title='Normalize BIST 30 ve Normalize Kur (2021)',
        labels = dict({'Date':'Tarih', 'value': 'kapanış (TRY)'}),
        )

In [20]:
#go grafigi

fig = go.Figure()
fig.add_trace(go.Bar(x = b30kur_2021.index, y = b30kur_2021['Normalize BIST30'], name = 'Normalize BIST30'))
fig.add_trace(go.Scatter(x = b30kur_2021.index, y = b30kur_2021['Normalize kapanış (TRY)'], name = 'Normalize EUR/TRY'))

fig.update_layout(title = 'Normalize BIST 30 ve Normalize Kur (2021)',
                  xaxis_title = 'Tarih', xaxis_dtick = 'M1',
                  yaxis_title = 'kapanış (TRY)')

### 2021 ve 2022 değerlerini birlikte inceleme

1. <i> Yahoo Finance</i> sitesinden 2022 BIST30 ve EUR/TRY kur değerlerini indiriniz

In [22]:
indirme_listesi = ['XU030.IS', 'EURTRY=X']
b30eur22 = pd.DataFrame()

for i in indirme_listesi:
  data = yf.download(i, start = '2022-01-01', end = '2023-01-01')
  data = data['Adj Close'].rename(i)
  b30eur22 = pd.concat([b30eur22, data], axis = 1)

b30eur22

[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


XU030.IS   EURTRY=X
2022-01-03  2109.500000  15.189530
2022-01-04  2157.000000  14.763000
2022-01-05  2212.300049  15.071520
2022-01-06  2219.600098  15.445910
2022-01-07  2256.600098  15.434040
...                 ...        ...
2022-07-11          NaN  17.503019
2022-07-12          NaN  17.325560
2022-07-15          NaN  17.506651
2022-08-30          NaN  18.190140
2022-12-26          NaN  19.734200

[260 rows x 2 columns]

2. Indirdiğiniz verileri 2021 verilerini temizleyip düzenlediğiniz üzere hazırlayınız.

In [23]:
b30eur22.columns = ['BIST30', 'Kapanış (TRY)'] #sütun isimlendirme

In [24]:
b30eur22['Normalize BIST30'] = b30eur22['BIST30'] / b30kur_2021.iloc[0, 0] #dikkat paydada 2021 var
b30eur22['Normalize Kapanış (TRY)'] = b30eur22['Kapanış (TRY)'] / b30kur_2021.iloc[0, 2]  #dikkat paydada 2021 var
b30eur22 = b30eur22[b30eur22['BIST30'] > 0] #bist 30 olmayan satırlardan kurtulmak için
b30eur22

BIST30  Kapanış (TRY)  Normalize BIST30  \
2022-01-03  2109.500000      15.189530          1.276242   
2022-01-04  2157.000000      14.763000          1.304979   
2022-01-05  2212.300049      15.071520          1.338435   
2022-01-06  2219.600098      15.445910          1.342852   
2022-01-07  2256.600098      15.434040          1.365237   
...                 ...            ...               ...   
2022-12-23  5885.200195      19.791670          3.560530   
2022-12-27  5876.700195      19.815269          3.555388   
2022-12-28  5789.399902      19.805389          3.502571   
2022-12-29  5915.200195      19.867519          3.578680   
2022-12-30  5944.799805      19.937571          3.596588   

            Normalize Kapanış (TRY)  
2022-01-03                 1.668138  
2022-01-04                 1.621296  
2022-01-05                 1.655178  
2022-01-06                 1.696294  
2022-01-07                 1.694990  
...                             ...  
2022-12-23                 2.173552  
2022-12-27                 2.176144  
2022-12-28                 2.175059  
2022-12-29                 2.181882  
2022-12-30                 2.189575  

[251 rows x 4 columns]

3. 2021 ve 2022 veri tablolarını tek bir tablo haline getiriniz.

In [26]:
b30eur = pd.concat([b30kur_2021, b30eur22], axis = 0)
b30eur = b30eur[b30eur['BIST30'] > 0]
b30eur

BIST30  Normalize BIST30  kapanış (TRY)  \
2021-01-04  1652.900024          1.000000        9.10568   
2021-01-05  1640.400024          0.992438        9.08882   
2021-01-06  1657.699951          1.002904        9.08177   
2021-01-07  1678.199951          1.015306        9.00614   
2021-01-08  1694.400024          1.025107        8.99850   
...                 ...               ...            ...   
2022-12-23  5885.200195          3.560530            NaN   
2022-12-27  5876.700195          3.555388            NaN   
2022-12-28  5789.399902          3.502571            NaN   
2022-12-29  5915.200195          3.578680            NaN   
2022-12-30  5944.799805          3.596588            NaN   

            Normalize kapanış (TRY)  Kapanış (TRY)  Normalize Kapanış (TRY)  
2021-01-04                 1.000000            NaN                      NaN  
2021-01-05                 0.998148            NaN                      NaN  
2021-01-06                 0.997374            NaN                      NaN  
2021-01-07                 0.989068            NaN                      NaN  
2021-01-08                 0.988229            NaN                      NaN  
...                             ...            ...                      ...  
2022-12-23                      NaN      19.791670                 2.173552  
2022-12-27                      NaN      19.815269                 2.176144  
2022-12-28                      NaN      19.805389                 2.175059  
2022-12-29                      NaN      19.867519                 2.181882  
2022-12-30                      NaN      19.937571                 2.189575  

[489 rows x 6 columns]

4. Veriyi grafik olarak inceleyiniz:<br>
	• Tabloları px grafiği olarak görselleştiriniz.<br>
	• Tabloları go. grafiği olarak görselleştiriniz.<br>
	• Tabloları subplots grafiği olarak görselleştiriniz.

In [27]:
#px grafigi
fig_3 = px.line(b30eur, y = ['Normalize BIST30', 'Normalize Kapanış (TRY)'])

fig_3.update_layout(title='Normalize BIST 30 ve Normalize Kur (2021-2022)',
                  xaxis_title = 'Tarih', xaxis_dtick = 'M1',
                  yaxis_title = 'Kapanış (TRY)')

fig_3.show()

In [28]:
#go grafigi

fig = go.Figure()
fig.add_trace(go.Bar(x = b30eur.index, y = b30eur['Normalize BIST30'], name = 'Normalize BIST 30'))
fig.add_trace(go.Scatter(x = b30eur.index, y = b30eur['Normalize Kapanış (TRY)'], name = 'Normalize EUR/TRY'))

fig.update_layout(title = '2021-2022 Normalize BIST 30 ve Normalize Kur',
                  xaxis_title = 'Tarih', xaxis_dtick = 'M1',
                  yaxis_title = 'Kapanış (TRY)')
